src/renderer/vt/VtSequences.cpp

```diff
@@ -333,25 +372,42 @@ using namespace Microsoft::Console::Render;
 }
 
 // Method Description:
-// - Formats and writes a sequence to change the underline of the following text.
+// - Formats and writes a sequence to change the extended underline styling of the following text.
+// - Uses backward compatible SGR 4 (without sub parameter) and SGR 21 for single and doubly underline.
 // Arguments:
-// - isUnderlined: If true, we'll underline the text. Otherwise we'll remove the underline.
+// - style: underline style to use.
 // Return Value:
 // - S_OK if we succeeded, else an appropriate HRESULT for failing to allocate or write.
-[[nodiscard]] HRESULT VtEngine::_SetUnderlined(const bool isUnderlined) noexcept
+[[nodiscard]] HRESULT VtEngine::_SetUnderlineExtended(const UnderlineStyle style) noexcept
 {
-    return _Write(isUnderlined ? "\x1b[4m" : "\x1b[24m");
+    switch (style)
+    {
+    case UnderlineStyle::NoUnderline:
+        return _SetUnderlined(false);
+    case UnderlineStyle::SinglyUnderlined:
+        return _SetUnderlined(true);
+    case UnderlineStyle::DoublyUnderlined:
+        return _Write("\x1b[21m");
+    case UnderlineStyle::CurlyUnderlined:
+        return _Write("\x1b[4:3m");
+    case UnderlineStyle::DottedUnderlined:
+        return _Write("\x1b[4:4m");
+    case UnderlineStyle::DashedUnderlined:
+        return _Write("\x1b[4:5m");
+    default:

```

Comment:
I'm in support of @j4james' suggestion to handle unknown combinations or unknown values as "singly underlines" here and in `s_GetGridLines`! It carries the _meaning_ (that there is _an_ underline, even if it is not the _right_ underline) better than not having an underline.

WDYT?

Reply:
Sounds right! I've pushed the changes, also added some comments over `GetUnderlineStyle(...)` so it's indicative that it might return invalid style values in certain cases.